sparse/src/KokkosSparse_Utils.hpp

```diff
@@ -2330,6 +2330,163 @@ void validateCrsMatrix(int m, int n, const Rowmap &rowmapIn,
   }
 }
 
+/**
+ * @brief Extract the diagonal blocks out of a crs matrix.
+ * This is a blocking function that runs on the host.
+ *
+ * @tparam crsMat_t The type of the CRS matrix
+ * @param A [in] The CrsMatrix.
+ * @param DiagBlk_v [out] The vector of extracted the CRS diagonal blocks.
```

Comment:
```suggestion
 * @param DiagBlk_v [out] The vector of the extracted CRS diagonal blocks.
```
Does anything else need to be stated here? Any assumptions (e.g. ordering of col idxs)? How does the caller specify the number of diagonal blocks to extract?